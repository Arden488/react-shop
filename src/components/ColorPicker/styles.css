.color-picker {
  display: grid;
  grid-template-columns: repeat(4, auto);
  grid-column-gap: 20px;
}

.color-picker__item {
  border: 1px solid #666;
  border-radius: 4px;
  min-height: 80px;
  text-align: center;
}

.color-picker__color {
  width: 20px;
  height: 20px;
  display: block;
  margin: 0 auto 10px;
  border-radius: 50%;
  background-color: #000;
}

.color-picker__color_gray {
  background-color: rgb(80, 80, 80);
}

.color-picker__item_selected {
  border-color: #ff9900;
}

.color-picker__label {
  font-size: 16px;
  display: block;
  margin: auto;
}

.color-picker__input {
  display: none;
}
