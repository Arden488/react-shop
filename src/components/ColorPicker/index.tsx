import * as React from 'react';

import './styles.css';

const ColorPicker = () => (
  <div className="color-picker">
    <button type="button" className="color-picker__item color-picker__item_selected">
    <span className="color-picker__color color-picker__color_gray" />
      <span className="color-picker__label">Gray space</span>
      <input className="color-picker__input" type="checkbox" name="" />
    </button>
    <button type="button" className="color-picker__item">
      <span className="color-picker__color" />
      <span className="color-picker__label">Black onyx</span>
      <input className="color-picker__input" type="checkbox" name="" />
    </button>
  </div>
);

export default ColorPicker;
