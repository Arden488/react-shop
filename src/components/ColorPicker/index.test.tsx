import { shallow } from 'enzyme';
import * as React from 'react';

import ColorPicker from './index';

let component: React.ReactNode = null;

beforeEach(() => {
  component = shallow(<ColorPicker />);
});

it('renders correctly', () => {
  expect(component).toMatchSnapshot();
});
