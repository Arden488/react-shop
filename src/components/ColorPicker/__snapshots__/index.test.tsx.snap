// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="color-picker"
>
  <button
    className="color-picker__item color-picker__item_selected"
    type="button"
  >
    <span
      className="color-picker__color color-picker__color_gray"
    />
    <span
      className="color-picker__label"
    >
      Gray space
    </span>
    <input
      className="color-picker__input"
      name=""
      type="checkbox"
    />
  </button>
  <button
    className="color-picker__item"
    type="button"
  >
    <span
      className="color-picker__color"
    />
    <span
      className="color-picker__label"
    >
      Black onyx
    </span>
    <input
      className="color-picker__input"
      name=""
      type="checkbox"
    />
  </button>
</div>
`;
